CONTRIBUTING.md

```diff
@@ -120,14 +122,361 @@ rework your branch. You should rework any relevant information from the GitHub
+  > For further details on golang, look at the
+  > [requirements section of the Kata Developer Guide](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#requirements-to-build-individual-components).
+
+
+- [Clone the upstream repository](https://help.github.com/articles/cloning-a-repository):
+
+  "Cloning" is a term that means to create a *local copy* of a git repository
+  that generally lives on a remote server.
+
+  ```sh
+  $ dir="$GOPATH/src/github.com/kata-containers"
+  $ mkdir -p "$dir"
+  $ cd "$dir"
+  $ git clone https://github.com/kata-containers/community
+  $ cd community
+  ```
+
+  > **Note:** The `git` command above has done the following:
+  >
+  > - Created a *local copy* of the upstream repository and put it in a *branch*
+  >   called `master`.
+  > - Created an "alias" (name) for the upstream repository URL called `origin`.
+
+  There are now *three* copies of the repository (all *exactly* the same):
+
+  - The original upstream one (https://github.com/kata-containers/community)
+
+    You do not (and cannot) make changes to this repository. PRs are a way of
+    allowing the project developers and administrators to review your work and
+    decide if they want to add your work to the upstream branch.
+
+    > **Note:** Don't worry - even if you make a mistake,
+    > you *cannot* do any damage to this repository.
+
+  - Your fork (`https://github.com/${your-github-username}/community`) on GitHub
+
+    This is where you will send or upload (`git push`) your PR branches.
+
+  - Your *local copy* of the upstream repository
+    (https://github.com/kata-containers/community)
+
+    This is where you will actually make changes to your private copy of the
+    repository.
+
+- [Setup a "git remote"](https://help.github.com/articles/configuring-a-remote-for-a-fork):
+
+  "But what happened to my fork?" you may be wondering. Good question! We're
+  about to use it...
+
+  Recall that your fork on the GitHub server is an *exact copy* of the
+  upstream repository that also lives on the GitHub server. Strictly, that may
+  not be true in the future - PRs are constantly being merged into the
+  upstream repository so at some point, your fork will become out of date
+  (it won't contain the latest changes in the upstream repository). The same
+  problem also affects your local copy of the upstream repository.
+
+  The way you keep your local copy up to date is to run `git pull`
+  periodically. This command will download all the latest changes from "another
+  repository" into your local repository. So how does `git` know which
+  repository to pull changes from? In simple terms it will default to pulling
+  from the repository you `git clone`'d above. Using `git` terminology, that
+  default repository is in fact given *two* names:
+
+  - The URL for the default upstream repository is called the `origin`.
+  - The for the default upstream repository is called the `origin`.
```

Comment:
Nit: "The fork of the default upstream repository"
Btw, nice explanation. If you would have written a book on it, I would have bought it. ;-)